synapse/storage/events.py

```diff
@@ -442,6 +433,9 @@ def _persist_events_txn(self, txn, events_and_contexts, backfilled):
             # Handle the case of the list including the same event multiple
             # times. The tricky thing here is when they differ by whether
             # they are an outlier.
+            if context.rejected:
```

Comment:
this code appears to have landed between a comment and the code it refers to, thus making the comment confusing. Because this method wasn't confusing enough before, I suppose.